src/test/components/TrackContextMenu.test.js

```diff
@@ -247,4 +248,52 @@ describe('timeline/TrackContextMenu', function() {
       // TODO - We should wait until we have some real non-thread tracks
     });
   });
+
+  describe('global / local track visibility interplay', function() {
+    function setupTracks() {
+      const results = setup();
+      const { getByText, dispatch, getState } = results;
+
+      const trackIndex = 1;
+      const trackReference = {
+        type: 'global',
+        trackIndex: trackIndex,
+      };
+      const track = getGlobalTracks(getState())[trackIndex];
+      if (track.type !== 'process') {
+        throw new Error('Expected a process track.');
+      }
+      const threadIndex = ensureExists(
+        track.mainThreadIndex,
+        `Couldn't get the mainThreadIndex of global track`
+      );
+
+      dispatch(changeSelectedThread(threadIndex));
+      dispatch(changeRightClickedTrack(trackReference));
+
+      const globalTrackItem = () => getByText('Content Process');
+      const localTrackItem = () => getByText('DOM Worker');
+
+      return {
+        ...results,
+        globalTrackItem,
+        localTrackItem,
+      };
+    }
+
+    it('will unhide the global track when unhiding one of its local tracks', function() {
+      const { getState, globalTrackItem, localTrackItem } = setupTracks();
+      // Hide the global track
```

Comment:
It might be nice to put one of these expect statements between each click so that readers can follow along with what the expectation of the behavior is.

```
      expect(getHumanReadableTracks(getState())).toEqual([
        'show [thread GeckoMain process] SELECTED',
        'show [thread GeckoMain tab]',
        '  - show [thread DOM Worker]',
        '  - hide [thread Style]',
      ]);
```